platforms/core-runtime/launcher/src/integTest/groovy/org/gradle/launcher/daemon/SingleUseDaemonIntegrationTest.groovy

```diff
@@ -101,6 +101,27 @@ assert java.lang.management.ManagementFactory.runtimeMXBean.inputArguments.conta
         daemons.daemon.stops()
     }
 
+    @Requires(IntegTestPreconditions.JavaHomeWithDifferentVersionAvailable)
+    def "forks build with default daemon JVM args when daemon jvm criteria from build properties does not match current process"() {
+        def otherJdk = AvailableJavaHomes.differentVersion
+        def otherJdkMetadata = AvailableJavaHomes.getJvmInstallationMetadata(otherJdk)
+
+        file('gradle/gradle-build.properties') << """
+            daemon.jvm.toolchain.version=$otherJdkMetadata.languageVersion.majorVersion
+            daemon.jvm.toolchain.vendor=$otherJdkMetadata.vendor.knownVendor
+        """
+
+        file('build.gradle') << "println 'javaVersion=' + org.gradle.internal.jvm.Jvm.current().javaVersion"
+
+        when:
+        succeeds()
+
+        then:
+        wasForked()
+        outputContains("javaVersion=$otherJdk.javaVersion")
+        daemons.daemon.stops()
```

Comment:
```suggestion
        file('build.gradle') << "assert org.gradle.internal.jvm.Jvm.current().javaVersion.toString() == '${otherJdk.javaVersion}'"

        when:
        succeeds()

        then:
        wasForked()
        daemons.daemon.stops()
```

Instead of doing a `println` + `outputContains`, can we switch this to an assertion?